### PR TITLE
#270 - Add support for retaining existing property values if overriding.

### PR DESCRIPTION
**Pull request by treshugart**

### CODE REVIEW

File: src/lifecycle/properties.js [hunk 1]
```diff
@@ -122,6 +123,8 @@ function defineProperty (elem, name, prop) {
       elem.attributeChangedCallback(prop.attr, null, elem.getAttribute(prop.attr));
     } else if (prop.init) {
       elem[name] = prop.init.call(this);
+    } else if (existingValue !== undefined) {
```
Comment (alexreardon) [inline src/lifecycle/properties.js]: :+1:

File: test/unit/lifecycle/properties.js [hunk 2]
```diff
@@ -218,4 +218,36 @@ describe('lifecycle/properties', function () {
     el.prop1 = '';
     expect(el.getAttribute('prop1')).to.equal('');
   });
+
+  describe('270', function () {
```
Comment (alexreardon) [inline test/unit/lifecycle/properties.js]: i think this should have a more descriptive name of the issue under test. Perhaps:

`Existing properties`

File: src/lifecycle/properties.js [hunk 3]
```diff
@@ -122,6 +123,8 @@ function defineProperty (elem, name, prop) {
       elem.attributeChangedCallback(prop.attr, null, elem.getAttribute(prop.attr));
     } else if (prop.init) {
       elem[name] = prop.init.call(this);
```
Comment (alexreardon) [inline src/lifecycle/properties.js]: Should init take precedence over what is on the element?

File: src/lifecycle/properties.js [hunk 4]
```diff
@@ -120,6 +121,8 @@ function defineProperty (elem, name, prop) {
   return function () {
     if (prop.attr && elem.hasAttribute(prop.attr)) {
       elem.attributeChangedCallback(prop.attr, null, elem.getAttribute(prop.attr));
+    } else if (existingValue !== undefined) {
```
Comment (alexreardon) [inline src/lifecycle/properties.js]: :+1:

**Comment by alexreardon** [comment]
PR looks good to go @treshugart

File: test/unit/lifecycle/properties.js (hunk 2 above)
Comment (treshugart) [inline test/unit/lifecycle/properties.js]: The sub tests will be appended to that, that's just to classify it. Could do that tho.

**Comment by alexreardon** [comment]
I was thinking that perhaps instead of calling it `init` it should be called `fallback` (ideally `default` but that is reserved word) to be clearer about it's behaviour

**Pull request merged**